x/emissions/keeper/keeper.go

```diff
@@ -3253,6 +3253,66 @@ func (k *Keeper) TopicExists(ctx context.Context, topicId TopicId) (bool, error)
 	return k.topics.Has(ctx, topicId)
 }
 
+// UpdateTopic applies allowed changes to a topic.
+func (k *Keeper) UpdateTopic(ctx context.Context, msg *types.UpdateTopicRequest) (types.Topic, error) {
+	params, err := k.GetParams(ctx)
+	if err != nil {
+		return types.Topic{}, errorsmod.Wrapf(err, "error getting params for sender: %v", &msg.Sender)
+	}
+
+	topic, err := k.GetTopic(ctx, msg.TopicId)
+	if err != nil {
+		return types.Topic{}, errorsmod.Wrapf(err, "error getting topic: %v", msg.TopicId)
+	}
+
+	if topic.Creator != msg.Sender {
+		return types.Topic{}, types.ErrNotPermittedToModifyTopic
```

Comment:
Maybe we can use the generic unauthorized error here:

```suggestion
		return types.Topic{}, sdkerrors.ErrUnauthorized
```

It's easier for clients to identify the underlying error when the code for a same error is consistent across modules